lua/hop/jump_target.lua

```diff
@@ -393,6 +393,18 @@ function M.regex_by_line_start()
   }
 end
 
+-- Line regex at cursor position.
+function M.regex_by_line_position()
+  local position = vim.api.nvim_win_get_cursor(0)[2]
+  local pattern = vim.regex(string.format("^.\\{0,%d\\}\\(.\\|$\\)", position))
```

Comment:
May you explain a bit the pattern?

Reply:
Sure, the idea is to locate the current cursor position, but fallback to the end of the line if the line is shorter.
I implemented it to match with how the other hint functions work (i.e. with regex), my solution was to:

  1. Match any character from the start, up to the current cursor position (`^.\\{0,%d\\}`), and
  2. Either the character after, or the end of the line (`\\(.\\|$\\)`).

`2` works because, if the line is shorter, `1` will still match. 
I originally wanted to implement an alternative solution, but this works using the same way the hint functions currently operate. 
Thus, resulting in minimal code changes whilst still achieving the desired result.

Reply:
Thanks for the explanation. I think we can merge your command now. :)